Makefile.win

```diff
@@ -0,0 +1,233 @@
+MV = move /Y $1 $2
+RM = if exist $1 del /F /Q $1
+RMDIR = if exist $1 rd /S /Q $1
+
+O := .build
+SOURCES := $(call GLOB,src\\*.cr)
+SPEC_SOURCES := $(call GLOB,spec\\*.cr)
+override FLAGS += -D strict_multi_assign $(if $(release),--release )$(if $(stats),--stats )$(if $(progress),--progress )$(if $(threads),--threads $(threads) )$(if $(debug),-d )$(if $(static),--static )$(if $(LDFLAGS),--link-flags="$(LDFLAGS)" )$(if $(target),--cross-compile --target $(target) )$(if $(interpreter),,-Dwithout_interpreter )
+SPEC_WARNINGS_OFF := --exclude-warnings spec\std --exclude-warnings spec\compiler --exclude-warnings spec\primitives
+SPEC_FLAGS := $(if $(verbose),-v )$(if $(junit_output),--junit_output $(junit_output) )
+CRYSTAL_CONFIG_LIBRARY_PATH := $$ORIGIN\lib
+CRYSTAL_CONFIG_BUILD_COMMIT := $(shell git rev-parse --short HEAD)
+CRYSTAL_CONFIG_PATH := $$ORIGIN\src
+SOURCE_DATE_EPOCH := $(shell git show -s --format=%ct HEAD)
+export_vars = $(eval export CRYSTAL_CONFIG_BUILD_COMMIT CRYSTAL_CONFIG_PATH SOURCE_DATE_EPOCH)
+export_build_vars = $(eval export CRYSTAL_CONFIG_LIBRARY_PATH)
+LLVM_CONFIG ?=
+LLVM_VERSION := $(if $(LLVM_CONFIG),$(shell $(LLVM_CONFIG) --version))
+LLVM_EXT_DIR = src\llvm\ext
+LLVM_EXT_OBJ = $(LLVM_EXT_DIR)\llvm_ext.obj
+DEPS = $(LLVM_EXT_OBJ)
+CXXFLAGS += $(if $(debug),/MTd /Od,/MT)
+CRYSTAL_VERSION ?= $(shell type src\VERSION)
+
+DESTDIR ?=
+prefix ?= $(if $(ProgramW6432),$(ProgramW6432),$(ProgramFiles))\crystal
+BINDIR ?= $(DESTDIR)$(prefix)
+LIBDIR ?= $(DESTDIR)$(prefix)\lib
+SRCDIR ?= $(DESTDIR)$(prefix)\src
+DATADIR ?= $(DESTDIR)$(prefix)
+
+colorize = $1
+
+check_llvm_config = $(eval \
+	check_llvm_config := $(if $(LLVM_VERSION),\
+		$(info $(call colorize,Using $(LLVM_CONFIG) [version=$(LLVM_VERSION)])),\
+		$(error "Could not locate compatible llvm-config, make sure it is installed and in your PATH, or set LLVM_CONFIG. Compatible versions: $(shell type src\llvm\ext\llvm-versions.txt)))\
+	)
+
+.PHONY: all
+all: crystal ## Build all files (currently crystal only) [default]
+
+define SP
+  
+endef
+
+.PHONY: help
+help: ## Show this help
+	@setlocal EnableDelayedExpansion &\
+	echo. &\
+	echo targets: &\
+	(for /F "usebackq tokens=1* delims=:" %%g in ($(MAKEFILE_LIST)) do (\
+		if not "%%h" == "" (\
+			set "_line=%%g                " &\
+			set "_rest=%%h" &\
+			set "_comment=!_rest:* ## =!" &\
+			if not "!_comment!" == "!_rest!"\
+				if "!_line:_rest=!" == "!_line!"\
+					echo   !_line:~0,16!!_comment!\
+		)\
+	)) &\
+	echo. &\
+	echo optional variables: &\
+	(for /F "usebackq tokens=1,3 delims=?#" %%g in ($(MAKEFILE_LIST)) do (\
+		if not "%%h" == "" (\
+			set "_var=%%g              " &\
+			echo   !_var:~0,14! %%h\
+		)\
+	)) &\
+	echo. &\
+	echo recipes: &\
+	(for /F "usebackq tokens=* delims=" %%g in ($(MAKEFILE_LIST)) do (\
+		set "_line=%%g" &\
+		if "!_line:~0,7!" == "##   $$ " (\
+			echo !_name! &\
+			echo  !_line:~2!\
+		) else if "!_line:~0,3!" == "## "\
+			set "_name=  !_line:~3!"\
+	))

```

Comment:
I'd prefer to place the `help` recipe at the end of the file. It's least relevant.